deps.bzl

```diff
@@ -1,10 +1,23 @@
-load("//private:version_repo.bzl", "version_repo")
+load("@bazel_tools//tools/build_defs/repo:http.bzl", "http_archive")
+load("@bazel_tools//tools/build_defs/repo:utils.bzl", "maybe")
 load("//private:globals.bzl", "GLOBALS")
 load("//private:globals_repo.bzl", "globals_repo")
+load("//private:version_repo.bzl", "version_repo")
 
-def bazel_features_deps():
+def bazel_features_deps(legacy = True):
```

Comment:
We can drop this parameter, there's essentially no cost to a module extension instantiating an unused repo.

Reply:
I tried removing the parameter. However, when I do, it errors under bzlmod mode:

```sh
ERROR: /Users/chuck/code/cgrindel/bazel_features/debug_stardoc_error/private/BUILD.bazel:14:12: in deps attribute of bzl_library rule //private:util: '@@_main~version_extension~bazel_features_version//:versio
n' does not have mandatory providers: 'StarlarkLibraryInfo'
ERROR: /Users/chuck/code/cgrindel/bazel_features/debug_stardoc_error/private/BUILD.bazel:14:12: Analysis of target '//private:util' failed
ERROR: Analysis of target '//private:util' failed; build aborted
INFO: Elapsed time: 0.283s, Critical Path: 0.00s
INFO: 1 process: 1 internal.
ERROR: Build did NOT complete successfully
ERROR: No test targets were found, yet testing was requested
```

Reply:
That's a very unexpected error condition. Are we getting both a workspace and a bzlmod Stardoc repo? I would like to understand this failure better before we merge this.

Reply:
I don't understand your concern with the boolean argument. The following is my understanding how `stardoc` is loaded:

- In bzlmod mode, `stardoc` is loaded from the `MODULE.bazel`. Hence, we do not need/want to load it again when the `version_extension` calls `bazel_features_deps`.
- In legacy mode, the client will call `bazel_features_deps` in their WORKSPACE. In this case, we do want `stardoc` to be declared, if they have not already done so.

I apologize if I am misunderstanding your concern.

Reply:
Are you referring to skylib (Stardoc is only a dev dependency)?

You are right, we do need skylib in the dependency macro for WORKSPACE users. I'm just saying that we don't need to introduce a parameter for the macro (adding user-facing complexity). Instead, we can either accept that the module extension produces an unused repo or factor out the repos needed by Bzlmod into a separate macro (that isn't public API) called by the WORKSPACE macro.

Reply:
@fmeum I moved the repo creation to a separate function. PTAL.